tools/wadlto/jaxrs/src/main/java/org/apache/cxf/tools/wadlto/jaxrs/SourceGenerator.java

```diff
@@ -1203,20 +1211,26 @@ private void writeJaxrResponse(StringBuilder sbCode, Set<String> imports) {
         sbCode.append(Response.class.getSimpleName()).append(" ");
     }
 
-    private Element getOKResponse(List<Element> responseEls) {
-        for (int i = 0; i < responseEls.size(); i++) {
-            String statusValue = responseEls.get(i).getAttribute("status");
-            if (statusValue.length() == 0) {
-                return responseEls.get(i);
-            }
-            String[] statuses = statusValue.split("\\s");
-            for (String status : statuses) {
-                if (HTTP_OK_STATUSES.contains(status)) {
-                    return responseEls.get(i);
+    private static Element getOKResponse(List<Element> responseEls) {
+        final List<Element> result = getResponses(responseEls, HTTP_OK_STATUSES);
```

Comment:
One problem with sharing the getResponses() method is that we are creating a list for the OK case which is not really required, as we are only ever returning the first element.

Reply:
previously a first OK element returned; now all OK elements collected and the same first element returned - so getOKResponse behaviour is the same

Reply:
The behaviour is the same, but we are creating a list just to store one element for the OK case.